render.c

```diff
@@ -274,6 +280,19 @@ void render_frame(struct swaylock_surface *surface) {
 			cairo_set_source_u32(cairo, state->args.colors.layout_text);
 			cairo_show_text(cairo, layout_text);
 			cairo_new_sub_path(cairo);
+
+			new_height += fe.height + 2 * box_padding;
+			if (new_width < (extents.width + 2 * box_padding)) {
+				new_width = extents.width + 2 * box_padding;
+			}
+		}
+
+		if (buffer_width != new_width || buffer_height != new_height) {
+			destroy_buffer(surface->current_buffer);
+			surface->current_buffer->width = new_width;
```

Comment:
Hijacking `current_buffer` to store the current width/height is a little strange to me. Maybe we can add some `indicator_{width,height}` fields instead?